fl4health/utils/losses.py

```diff
@@ -2,15 +2,103 @@
 
 import torch
 
 
-class Losses:
+class Losses(ABC):
+    def __init__(self, additional_losses: Optional[Dict[str, torch.Tensor]] = None) -> None:
+        """
+        An abstract class to store the losses
+
+        Args:
+            additional_losses (Optional[Dict[str, torch.Tensor]]): Optional dictionary of additional losses.
+        """
+        self.additional_losses = additional_losses if additional_losses else {}
+
+    def as_dict(self) -> Dict[str, float]:
+        """
+        Produces a dictionary representation of the object with all of the losses.
+
+        Returns:
+            Dict[str, float]: A dictionary with the additional losses if they exist.
+        """
+        loss_dict: Dict[str, float] = {}
+
+        if self.additional_losses is not None:
+            for key, val in self.additional_losses.items():
+                loss_dict[key] = float(val.item())
+
+        return loss_dict
+
+    @staticmethod
+    @abstractmethod
+    def aggregate(loss_meter: LossMeter) -> Losses:
+        """
+        Aggregates the losses in the given LossMeter into an instance of Losses
+
+        Args:
+            loss_meter (LossMeter): The loss meter object with the collected losses.
+
+        Raises:
+            NotImplementedError: To be implemented by child classes.
+        """
+        raise NotImplementedError
+
+
+class EvaluationLosses(Losses):
+    def __init__(self, checkpoint: torch.Tensor, additional_losses: Optional[Dict[str, torch.Tensor]] = None) -> None:
+        """
+        A class to store the checkpoint and additional_losses of a model
+        along with a method to return a dictionary representation.
+
+        Args:
+            checkpoint (torch.Tensor): The loss used to checkpoint model (if checkpointing is enabled).
+            additional_losses (Optional[Dict[str, torch.Tensor]]): Optional dictionary of additional losses.
+        """
+        super().__init__(additional_losses)
+        self.checkpoint = checkpoint
+
+    def as_dict(self) -> Dict[str, float]:
+        """
+        Produces a dictionary representation of the object with all of the losses.
+
+        Returns:
+            Dict[str, float]: A dictionary with the checkpoint loss, plus each one of the keys in
+                additional losses if they exist.
+        """
+        loss_dict = super().as_dict()
+        loss_dict["checkpoint"] = float(self.checkpoint.item())
+        return loss_dict
+
+    @staticmethod
+    def aggregate(loss_meter: LossMeter) -> EvaluationLosses:
+        """
+        Aggregates the losses in the given LossMeter into an instance of EvaluationLosses
+
+        Args:
+            loss_meter (LossMeter): The loss meter object with the collected losses.
+
+        Returns:
+            EvaluationLosses: An instance of EvaluationLosses with the aggregated losses.
+        """
+        assert all([isinstance(losses, EvaluationLosses) for losses in loss_meter.losses_list])
```

Comment:
I think, though I'm not certain, that if we generically type the LossMeter, we can also drop this? I'm not 100% sure though.

Reply:
Yes, I think this can be dropped once we do the generics change, which is also one more reason to do it.